scripts/leapp_preupgrade.py

```diff
@@ -0,0 +1,340 @@
+import json
+import os
+import subprocess
+
+
+# Both classes taken from:
+# https://github.com/oamg/convert2rhel-worker-scripts/blob/main/scripts/preconversion_assessment_script.py
+class ProcessError(Exception):
+    """Custom exception to report errors during setup and run of leapp"""
+
+    def __init__(self, message):
+        super(ProcessError, self).__init__(message)
+        self.message = message
```

Comment:
In this current release, @Wecros, it needs to be self-contained. We don't have the support for split files. 

Is that a feature that you envision somehow? If so, we can talk more about that sometime.

Reply:
Alright thanks, understood. Sure we can talk about that, but it's not a problem for me, I was more probing for the implementation limitations and what we can do when it comes to code design.

Reply:
I don't think it would be a great feature, it would make whole worker implementation more complicated, we would somehow have to download multiple files (I guess archive?) and either insights or our worker would have to extract it - overall seems like too much effort for little benefit. There would also be the questions about signing such archive because you have to remember that this is raw script but insights serves it signed inside yaml envelope.